apiserver/common/secrets/secrets.go

```diff
@@ -604,21 +604,34 @@ func removeSecrets(
 
 	removeFromExternal := func(uri *coresecrets.URI, revisions ...int) error {
 		externalRevs := make(map[string]provider.SecretRevisions)
-		for _, rev := range revisions {
-			revMeta, err := removeState.GetSecretRevision(uri, rev)
-			if err != nil {
-				return errors.Trace(err)
-			}
-			valRef := revMeta.ValueRef
+		addExternalRevs := func(valRef *coresecrets.ValueRef) {
```

Comment:
Can we call this "gatherExternalRevs"
"add" make me look twice since we were in a remove function